### ISSUE TITLE
Syntax highlighting breaks after interpolation

### ISSUE THREAD

**Issue by alloy** [kumarharsh/graphql-for-vscode#20]
Highlighting breaks after an interpolated string:

<img width="523" alt="screen shot 2017-03-04 at 15 33 43" src="https://cloud.githubusercontent.com/assets/2320/23579548/1ea72f22-00f0-11e7-81e0-09c56b65ad3f.png">

This is what the rest of the file should look like:

<img width="540" alt="screen shot 2017-03-04 at 15 33 08" src="https://cloud.githubusercontent.com/assets/2320/23579553/27111d76-00f0-11e7-8a0f-8c4f6b71b57d.png">

----

I _think_ this thread describes that it has been fixed in their grammar, but not sure: https://github.com/gandm/language-babel/issues/242

**Comment by kumarharsh** [comment]
@alloy make sure you've chosen 'Javascript React' as the language. Also, if possible, try installing the 'Sublime Babel' plugin by Josh Peng. If that doesn't help, can you share the plugins you've installed? I'm not getting this error:

![image](https://cloud.githubusercontent.com/assets/760112/23580068/79b6e882-0120-11e7-98c2-fa8b14201f68.png)

**Comment by alloy** [comment]
Sorry, should have mentioned this is TypeScript and given you a snippet, in case you still need it you can find an example file [here](https://github.com/artsy/reaction-force/blob/bb847e8fac8567ab7a1fa3a5d89d495f998c843f/src/components/__stories__/artwork_grid.tsx).

I tried switching to JavaScript React, but the same problem occurs. However, I did find the real cause, it is because I use double-quote and you use single-quote. Can you try updating your example to double-quotes and see if you can reproduce it that way?

**Comment by kumarharsh** [comment]
@alloy, wow, yes, double quotes don't work apparently. will look into it.

**Comment by kumarharsh** [comment]
This was fixed in 4957092ae9424c7bd8e570495c3df6423de683ce. Let me know if you face the issue again.

**Comment by alloy** [comment]
@kumarharsh Awesome, that does the trick!

However, I did notice that I now lost JS syntax highlighting for the string interpolation (both the interpolation syntax _and_ JS syntax inside of it). Is that a known thing (because I see it happening in your above screenshot too)?

## Before

<img width="275" alt="screen shot 2017-04-14 at 10 31 36" src="https://cloud.githubusercontent.com/assets/2320/25038182/9ce6fcc8-20fd-11e7-982c-21ccc296b16b.png">

## After

<img width="296" alt="screen shot 2017-04-14 at 10 31 48" src="https://cloud.githubusercontent.com/assets/2320/25038187/a1dae62c-20fd-11e7-8dd8-bbc98c2de2b7.png">

**Comment by kumarharsh** [comment]
@alloy that's surprising, because I've never gotten the strings to highlight at all for me, even with monokai theme XD

What I found was an intrinsic problem with how the patterns were defined, which you can read more about in the description of the commit I linked to above.

**Comment by alloy** [comment]
Hahaha ¯\_(ツ)_/¯

Shall I file a separate ticket for that issue?

**Comment by kumarharsh** [comment]
No, lets track it here itself.

**Comment by kumarharsh** [comment]
If possible, can you rebuild the extension before this commit, and tell me the scopes being applied to `ArtworkGrid`, `.getFragment` and `"artworks"`?

What I get right now for the first two are:

```
variable.graphql
meta.selectionset.graphql
meta.selectionset.graphql
meta.selectionset.graphql
meta.selectionset.graphql
meta.selectionset.graphql
meta.fragment.graphql
relayQL
meta.group.braces.curly.js
source.js
```

and the last one `"artworks"` doesn't have `variable.graphql`.

**Comment by alloy** [comment]
How can I tell what scopes are being applied?

**Comment by kumarharsh** [comment]
Ctrl-Shift-P > Inspect TM Scopes

**Comment by kumarharsh** [comment]
@alloy - I think I might have solved it - check out the changes at 6704c13

**Comment by alloy** [comment]
Aw yeah, works great! 👏 

(Sorry for not responding earlier.)
